src/composables/useTransactions.ts

```diff
@@ -0,0 +1,328 @@
+import { computed, ref } from 'vue';
+import { merge, orderBy } from 'lodash';
+import { TransactionReceipt } from '@ethersproject/providers';
+import { useI18n } from 'vue-i18n';
+
+import { configService } from '@/services/config/config.service';
+import { gnosisOperator } from '@/services/gnosis/operator.service';
+import useVueWeb3 from '@/services/web3/useVueWeb3';
+import { OrderMetaData } from '@/services/gnosis/types';
+import { gnosisExplorer } from '@/services/gnosis/explorer.service';
+
+import { lsGet, lsSet } from '@/lib/utils';
+
+import LS_KEYS from '@/constants/local-storage.keys';
+import useNotifications from './useNotifications';
+
+const DAY_MS = 86_400_000;
+
+export type TransactionStatus =
+  | 'pending'
+  | 'confirmed'
+  | 'expired'
+  | 'cancelling'
+  | 'cancelled';
+
+export type TransactionAction =
+  | 'claim'
+  | 'approve'
+  | 'trade'
+  | 'invest'
+  | 'withdraw';
+
+export type TransactionType = 'order' | 'tx';
+
+export type TxReceipt = Pick<
+  TransactionReceipt,
+  | 'blockHash'
+  | 'blockNumber'
+  | 'contractAddress'
+  | 'from'
+  | 'status'
+  | 'to'
+  | 'transactionHash'
+  | 'transactionIndex'
+>;
+
+export type Transaction = {
+  id: string;
+  action: TransactionAction;
+  type: TransactionType;
+  receipt?: OrderMetaData | TxReceipt;
+  details?: Record<string, any>;
```

Comment:
Passing in `details` will allow reconstructing an advanced preview that has pretty much all the info needed. This is different from the one-liner `summary` field which is a quick way of showing what happened. Each transaction can set its own `details` depending on the action (trade/approve/claim/invest...)

I do believe that once all transaction types are in I will be able to type it better than just `Record<string, any>`.